statics/js/components/preference.js

```diff
@@ -33,114 +33,97 @@ var PreferenceComponent = {
           <a><i class="fa fa-question help-text" aria-hidden="true" title="Warning threshold in kbps"></i></a>\
-          <input id="bw-abs-warning" type="number" class="form-control input-sm" v-model="bwAbsWarning" min="0"/>\
+          <input id="bw-abs-warning" type="number" class="form-control input-sm" v-model="preferences.bandwidthAbsoluteWarning" min="0"/>\
         </div>\
         <div class="form-group">\
           <label for="bw-abs-alert">Bandwidth Absolute Alert</label>\
           <a><i class="fa fa-question help-text" aria-hidden="true" title="Alert threshold in kbps"></i></a>\
-          <input id="bw-abs-alert" type="number" class="form-control input-sm" v-model="bwAbsAlert" min="0"/>\
+          <input id="bw-abs-alert" type="number" class="form-control input-sm" v-model="preferences.bandwidthAbsoluteAlert" min="0"/>\
         </div>\
       </div>\
-      <div v-if="bwThreshold == \'relative\'">\
+      <div v-if="preferences.bandwidthThreshold == \'relative\'">\
         <div class="form-group">\
           <label for="bw-rel-active">Bandwidth Relative Active</label>\
           <a><i class="fa fa-question help-text" aria-hidden="true" title="Active threshold in between 0 to 1"></i></a>\
-          <input id="bw-rel-active" type="number" class="form-control input-sm" v-model="bwRelActive" min="0" max="1" step="0.1"/>\
+          <input id="bw-rel-active" type="number" class="form-control input-sm" v-model="preferences.bandwidthRelativeActive" min="0" max="1" step="0.1"/>\
         </div>\
         <div class="form-group">\
           <label for="bw-rel-warning">Bandwidth Relative Warning</label>\
           <a><i class="fa fa-question help-text" aria-hidden="true" title="Warning threshold in between 0 to 1"></i></a>\
-          <input id="bw-rel-warning" type="number" class="form-control input-sm" v-model="bwRelWarning" min="0" max="1" step="0.1"/>\
+          <input id="bw-rel-warning" type="number" class="form-control input-sm" v-model="preferences.bandwidthRelativeWarning" min="0" max="1" step="0.1"/>\
         </div>\
         <div class="form-group">\
           <label for="bw-rel-alert">Bandwidth Relative Alert</label>\
           <a><i class="fa fa-question help-text" aria-hidden="true" title="Alert threshold in between 0 to 1"></i></a>\
-          <input id="bw-rel-alert" type="number" class="form-control input-sm" v-model="bwRelAlert" min="0" max="1" step="0.1"/>\
+          <input id="bw-rel-alert" type="number" class="form-control input-sm" v-model="preferences.bandwidthRelativeAlert" min="0" max="1" step="0.1"/>\
         </div>\
       </div>\
       <div class="button-holder row">\
-        <button class="btn btn-lg btn-primary" type="submit">Save</button>\
+        <button class="btn btn-lg btn-primary" type="submit" title="save in local storage">Save</button>\
+        <button class="btn btn-lg btn-primary" type="button" @click="saveToFile" title="download preferences to local file"> Download</button>\
```

Comment:
I would rather use Export/Import aligning these button to the left for ex and the save/cancel to the right with a space between the two groups. That said I had an issue, I changed the preferences, I downloaded, then I cancel and the "cancelled" preferences were applied.